core/src/main/java/net/opentsdb/query/processor/downsample/DownsampleFactory.java

```diff
@@ -288,10 +293,10 @@ public void setupGraph(final QueryPipelineContext context,
           new_source.addSummaryAggregation(newConfig.getAggregator());
         }
         plan.replace(source, new_source.build());
-      }      
+      }
+
+      plan.replace(config, newConfig);
```

Comment:
This seems to be replacing the given downsample config with the new one for every Tiemseries data source. In the end, we will choose the last seen Timeseries Data source config's start and end ?

Reply:
This needs a little cleanup. I've moved the replace outside now but there are still some extra bits needed to handle multiple sources properly.